### PR TITLE
Treat the return value of read_changeset_data as string

### PR DESCRIPTION
**Pull request by ehsan**
This fixes issue #23.

### CODE REVIEW

File: git-remote-hg [hunk 1]
```diff
@@ -332,7 +332,7 @@ class BranchMap(object):
                 sha1 = store.changeset_ref(head)
                 if not sha1:
                     continue
-                extra = store.read_changeset_data(sha1).get('extra')
+                extra = store.read_changeset_data(sha1)
```
Comment (glandium) [inline git-remote-hg]: This breaks the code that follows. But read_changeset_data shouldn't be returning None in the first place. What is the output of git cinnabar fsck?

Comment (glandium) [inline git-remote-hg]: Also, if line 332 returns a value, then read_changeset_data should return something. Your error suggests that the cinnabar metadata is inconsistent, and the question is how that happened.

**Comment by ehsan** [comment]
The output of git cinnabar fsck is rather large: http://people.mozilla.org/~eakhgari/cinnabar-log.txt

Not sure how I got myself into this state...

**Comment by glandium** [comment]
Ouch. Could you push refs/cinnabar/hg2git, refs/cinnabar/manifest and refs/notes/cinnabar to your gecko-dev repo?

**Comment by ehsan** [comment]
I pushed those refs from my recloned repo to my gecko-dev clone.

**Comment by glandium** [comment]
Thanks for the refs. See issue #23 for some details for the likely root cause. So, this patch is not going to help.

**Pull request closed without merging**